### ISSUE TITLE
URL bar does not show the current location (shows URLs of network requests loaded)

### ISSUE THREAD

**Issue by cvan** [MozillaReality/FirefoxReality#738]
## Hardware
Oculus Go

## Steps to Reproduce
1. Load https://blog.twitter.com/
2. Notice the URL bar changes to https://static.ads-twitter.com/oct.js

## Current Behavior
The URL bar changes for each network request that occurs during the current document load.

![fxr-url-wrong-bug](https://user-images.githubusercontent.com/203725/48232148-822d2980-e365-11e8-8cbb-c5fc25f05ca6.gif)

I looked in the WebIDE, and `location.href` is still `https://blog.twitter.com/`.

From `adb logcat`:

```
11-08 14:33:46.923  9842  9874 D GeckoViewNavigation[C]: loadURI: uri=https://blog.twitter.com/ where=1 flags=0x1 tp=null
```

```
11-08 14:34:29.532  9842  9874 D GeckoViewNavigation[C]: loadURI: uri=https://static.ads-twitter.com/oct.js where=1 flags=0 tp=null
```

## Expected Behavior
The URL bar shows the true URL, not the URIs for the network requests loading.

## Error Logs and Stack Traces
[**`adb logcat`** results](https://gist.github.com/cvan/06f76117640051be23529cc760a409b5#file-adb-logcat-md)

**Comment by cvan** [comment]
related: issue #617:

![image](https://user-images.githubusercontent.com/203725/48448293-6eae0400-e753-11e8-87d2-4c1dc670eebb.png)

**Comment by keianhzo** [comment]
Both things have been introduced in GeckoView recently as this didn't happened before. We should only update the URL bar when getting onPageStart and not on a onLoadRequest, that should solved both #617 and #738.
